autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/FeatureAgglomeration.py

```diff
@@ -0,0 +1,122 @@
+        n_clusters: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='n_clusters',
+                                                                          value_range=(0.5, 0.9),
+                                                                          default_value=0.5,
+                                                                          ),
+        affinity: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='affinity',
+                                                                        value_range=("euclidean",
+                                                                                     "manhattan",
+                                                                                     "cosine"),
+                                                                        default_value="euclidean",
+                                                                        ),
+        linkage: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='linkage',
+                                                                       value_range=("ward", "complete", "average"),
+                                                                       default_value="ward",
+                                                                       ),
+        pooling_func: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='pooling_func',
+                                                                            value_range=("mean", "median", "max"),
+                                                                            default_value="max",
+                                                                            ),
+    ) -> ConfigurationSpace:
+        if dataset_properties is not None:
+            n_features = len(dataset_properties['numerical_columns']) if isinstance(
+                dataset_properties['numerical_columns'], List) else 0
+            if n_features == 1:
+                log = False
+            else:
+                log = n_clusters.log
+            n_clusters = HyperparameterSearchSpace(hyperparameter='n_clusters',
+                                                   value_range=(
+                                                       floor(float(n_clusters.value_range[0]) * n_features),
+                                                       ceil(float(n_clusters.value_range[1]) * n_features)),
+                                                   default_value=ceil(float(n_clusters.default_value) * n_features),
+                                                   log=log)
+        else:
+            n_clusters = HyperparameterSearchSpace(hyperparameter='n_clusters',
+                                                   value_range=(2, 400),
+                                                   default_value=25,
+                                                   log=n_clusters.log)
```

Comment:
Can we make a separated method to get those arguments?

Reply:
done